pkg/driver/node.go

```diff
@@ -118,37 +119,62 @@ func (d *nodeService) NodePublishVolume(ctx context.Context, req *csi.NodePublis
 	}
 
 	klog.V(5).Infof("NodePublishVolume: mounting juicefs with secret %+v, options %v", reflect.ValueOf(secrets).MapKeys(), mountOptions)
-	jfs, err := d.juicefs.JfsMount(secrets, mountOptions)
+	jfs, err := d.juicefs.JfsMount(volumeID, secrets, mountOptions)
 	if err != nil {
 		return nil, status.Errorf(codes.Internal, "Could not mount juicefs: %v", err)
 	}
 
-	bindSource := jfs.GetBasePath()
+	var bindSource string
+	if m, ok := req.GetVolumeContext()["mode"]; ok && m == "dynamic" {
+		// dynamic provisioning
+		var capacity int64 = juicefs.DefaultCapacityBytes
+		if c, ok := req.GetVolumeContext()["capacity"]; ok {
+			if v, err := strconv.ParseInt(c, 10, 64); err == nil {
+				capacity = v
+			}
+		}
+
+		vol, err := jfs.CreateVol(volumeID, capacity)
+		if err != nil {
+			return nil, status.Errorf(codes.Internal, "Could not create volume: %s, %v", volumeID, err)
+		}
+
+		bindSource = vol.VolumePath
+	} else {
+		// static provisioning
+		bindSource = jfs.GetBasePath()
+	}
 	if subPath, ok := req.GetVolumeContext()["subPath"]; ok {
 		bindSource = path.Join(bindSource, subPath)
```

Comment:
if pre-existing directory is needed, `subPath` will be used, relative to root path